tests/e2e_test.go

```diff
@@ -279,84 +280,112 @@ func TestProxyAppliedOnExistingWorkload(t *testing.T) {
 	}
 }
 
-func TestPostgresConnection(t *testing.T) {
+func TestPublicDbConnections(t *testing.T) {
 	// When running tests during development, set the SKIP_CLEANUP=true envvar so that
 	// the test namespace remains after the test ends. By default, the test
 	// namespace will be deleted when the test exits.
 	skipCleanup := loadValue("SKIP_CLEANUP", "", "false") == "true"
-
-	ctx := testContext()
-
-	tp := newTestCaseClient("pgconnection")
-
-	err := tp.CreateOrPatchNamespace(ctx)
-	if err != nil {
-		t.Fatal(err)
-	}
-	t.Cleanup(func() {
-		if skipCleanup {
-			return
-		}
-
-		err = tp.DeleteNamespace(ctx)
-		if err != nil {
-			t.Fatal(err)
-		}
-	})
-
 	const (
 		pwlName  = "newss"
-		appLabel = "pgsql"
+		appLabel = "client"
 		kind     = "Deployment"
 	)
-	key := types.NamespacedName{Name: pwlName, Namespace: tp.Namespace}
-
-	s := testhelpers.BuildSecret("db-secret",
-		"DB_USER", "postgres",
-		"DB_PASS", tp.DBRootPassword,
-		"DB_NAME", tp.DBName)
-	s.SetNamespace(tp.Namespace)
-	err = tp.Client.Create(ctx, &s)
-	if err != nil {
-		t.Fatal(err)
+
+	tests := []struct {
+		name        string
+		c           *testhelpers.TestCaseClient
+		podTemplate corev1.PodTemplateSpec
+		allOrAny    string
+	}{
+		{
+			name:        "postgres",
+			c:           newPublicPostgresClient("postgresconn"),
+			podTemplate: testhelpers.BuildPgPodSpec(600, appLabel, "db-secret"),
+			allOrAny:    "all",
+		},
+		{
+			name:        "mysql",
+			c:           newPublicMysqlClient("mysqlconn"),
+			podTemplate: testhelpers.BuildMysqlPodSpec(600, appLabel, "db-secret"),
+			allOrAny:    "all",
+		},
+		{
+			name:        "mssql",
+			c:           newPublicMssqlClient("mssqlconn"),
+			podTemplate: testhelpers.BuildMSSQLPodSpec(600, appLabel, "db-secret"),
+			allOrAny:    "all",
+		},
 	}
+	for i := range tests {
```

Comment:
Why not `for i, tc := range tests { }` here?

Reply:
Let's try to resist doing this unless absolutely necessary. Is there any mutation here that makes it necessary? Also, have you tested if `t.Parallel` benefits the speed of the test runs here?

Reply:
Going forward, I'll write table tests like this: `for _, tc := range tests { }` unless there's a good reason to do otherwise. 

t.Parallel() is very helpful here. Each test case takes about 2 minutes, mostly waiting around for Kubernetes to reconcile and start pods.